template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/knox/KnoxIdBrokerConfigProvider.java

```diff
@@ -46,6 +51,15 @@
                     case AZURE:
                         config = List.of(config("idbroker_azure_user_mapping", userMappings),
                                 config("idbroker_azure_group_mapping", groupMappings));
+                        ClouderaManagerRepo cmRepo = source.getProductDetailsView().getCm();
+                        if (CMRepositoryVersionUtil.isIdBrokerManagedIdentitySupported(cmRepo)) {
+                            String idBrokerManagedIdentity = getIdBrokerManagedIdentity(source.getFileSystemConfigurationView());
+                            if (Objects.nonNull(idBrokerManagedIdentity)) {
+                                config = List.of(config("idbroker_azure_user_mapping", userMappings),
```

Comment:
Instead of duplicating the two mapping configs, I suggest changing lines 52--53 above to use a mutable list like `ArrayList` that the assumer identity config can be added to here. Though not strictly required, if we were to keep the immutable list convention, inserting something like `config = Collections.unmodifiableList(config);` before the `break` could work.